### ISSUE TITLE
internationnalisation (French localisation)

### ISSUE THREAD

**Issue by trebmuh** [Impro-Visor/Impro-Visor#37]
Hi all and thanks for Impro-visor, really nice software, congratz!

I was looking at a French translation of it and can't find one. So browsing the sources, I [found this](https://github.com/Impro-Visor/Impro-Visor/tree/master/src/imp/lang) where it looks like something was started around the 5.x version but has not been finished yet. Moreover, the few strings translated in French (at the beginning of what looks like a French localization file) doesn't show up in the 9.1 version that I'm using.

So I have a few questions:

1. Is there a mechanism already implemented to get a translated GUI?
2. If yes, is there a way to select a translation? (I've looked into the GUI preferences but didn't see nothing)
3. Would you be interested with me providing an effort to get a full French translation of it?

Cheers,
Olivier

**Comment by keller91711** [comment]
Yes, there was an effort to do a French version. No, it was never completed nor maintained.
Yes, we would be interested in a continued effort. Note that anything in this direction needs to be done in the context of the NetBeans IDE. If you are interested in adding French equivalents, I'm glad to try to help. I just can't be the main person on it.

Thanks.
Bob

**Comment by trebmuh** [comment]
@keller91711 thanks for the quick answer. Appreciated.
As I'm not coding-person, I'm not sure what you mean by:
> needs to be done in the context of the NetBeans IDE

Could you please elaborate a bit on this?

I'm am interested to add French translation equivalent and to keep it up to date when done (ie: translating new strings when a new version is close to be released).
After browsing the sources, I guess the answer to the next question is "no", but I might have miss something so, I'm still asking for it : is there another language translation already done that I could have a look to maybe?

**Comment by keller91711** [comment]
Netbeans is the software tool that we use for building a release from code. It helps with the graphic layouts and other things. Because we want to keep the system as a single source, it is necessary to add implement the internationalization in this tool. That way, when new releases are made, there will be a translated version for them. Otherwise, you would have to start again to keep up with new releases.

I'm hoping for the next release to be around the turn of the calendar. After that would be a good time to revisit the French translation version. There is no other language available right now.

**Comment by trebmuh** [comment]
Thanks for the explanation.

Please, fell free to ping me here when the next version is released then we can start to look around how to progress with the French translation.

In the meantime, enjoy the end of the year and the new one coming in soon.
